central/processlisteningonport/datastore/pruning.go

```diff
@@ -31,4 +31,9 @@ const (
 	// if the serialized data has process information. PLOPs without process information are then deleted.
 	getPotentiallyOrphanedPLOPs = `SELECT plop.serialized FROM listening_endpoints plop where NOT EXISTS
 			(select 1 FROM process_indicators proc where plop.processindicatorid = proc.id)`
+
+	// Finds PLOPs without poduids and deletes them. This is done in batches.
+	deletePLOPsWithoutPoduid = `WITH temp AS
+		(SELECT id, poduid FROM listening_endpoints LIMIT %d OFFSET %d)
```

Comment:
There could be an `ORDER BY` here to make sure this is more consistent and all rows are checked, but that would be more expensive.

Reply:
If the check for null poduid was here, this could scan the entire table after the offset, and do so multiple times.

Reply:
This query has been removed. I am using `ORDER BY` in other queries.